sdk/metric/view/view.go

```diff
@@ -17,22 +17,195 @@
+}
+
+func (v View) matchLibraryVersion(version string) bool {
+	return v.scope.Version == "" || version == v.scope.Version
+}
+
+func (v View) matchLibrarySchemaURL(schemaURL string) bool {
+	return v.scope.SchemaURL == "" || schemaURL == v.scope.SchemaURL
+}
 
-// Option applies a configuration option value to a view Config.
+// Option applies a Configuration option value to a view Config. All options
+// will be used together to determine match and transforms.
 type Option interface {
-	apply(Config) Config
+	apply(View) View
+}
+
+type optionFunc func(View) View
+
+func (f optionFunc) apply(v View) View {
+	return f(v)
+}
+
+// MatchInstrumentName will match an instrument based on the instruments name.
+// This will accept wildcards of * for zero or more characters, and ? for
+// exactly one character. A name of "*" (default) will match all instruments.
+func MatchInstrumentName(name string) Option {
+	return optionFunc(func(v View) View {
+		if strings.ContainsAny(name, "*?") {
+			v.hasWildcard = true
+		}
+		name = regexp.QuoteMeta(name)
+		name = "^" + name + "$"
+		name = strings.ReplaceAll(name, "\\?", ".")
+		name = strings.ReplaceAll(name, "\\*", ".*")
+		v.instrumentName = regexp.MustCompile(name)
+		return v
+	})
 }
 
-// TODO (#2837): implement view match options.
-// TODO (#2837): implement view annotation options.
+// TODO: Implement when InstrumentKind and NumberKind are defined
+// func MatchInstrumentKind(k sdkinstrument.Kind) Option {
+// 	return optionFunc(func(v Config) Config {
+// 		v.instrumentKind = k
+// 		return v
+// 	})
+// }
+
+// func MatchNumberKind(k number.Kind) Option {
+// 	return optionFunc(func(v Config) Config {
+// 		v.numberKind = k
+// 		return v
+// 	})
+// }
 
-// New returns a new and configured view Config.
-func New(opts ...Option) Config {
-	return Config{}
+// MatchInstrumentationLibrary will do an exact match on any
+// instrumentation.Library field that is not blank (""). Default is to match all
+// InstrumentationLibraries.
+func MatchInstrumentationLibrary(lib instrumentation.Library) Option {
+	return optionFunc(func(v View) View {
+		v.scope = lib
+		return v
+	})
 }
+
+// WithName will rename the instrument the view matches. If not used or empty the
+// instrument name will not be changed. Must be used with a non-wildcart
+// instrument name match. Default does no change.
+func WithName(name string) Option {
+	return optionFunc(func(v View) View {
+		v.viewName = name
+		return v
+	})
+}
+
+// WithDescription will change the description of the instruments the view
+// matches. If not used or empty the description will not be changed.
+func WithDescription(desc string) Option {
+	return optionFunc(func(v View) View {
+		v.description = desc
+		return v
+	})
+}
+
+// WithAttributeFilter will select attributes that have a matching key.  If not used
+// or empty no filter will be applied.
+func WithAttributeFilter(keys ...attribute.Key) Option {
```

Comment:
We have an `attribute.Filter` that it would make more sense to pass if we want to have an option `WithAttributeFilter`. Instead of using that name, can we use `WithAttributes` instead?

```suggestion
func WithAttributes(keys ...attribute.Key) Option {
```

Reply:
I'm ok with that, but we will have it confused with adding attributes to a stream.

Reply:
Gotcha, that might be confusing.

Since we have diverged in our option naming scheme with the `Match` options, does it make sense to name these closer to what they do? So for this it would be `FilterAttributes`, and the others would be something like `Rename` and `SetDescription`.

Reply:
Do we have any options in the project that don't have some kind of prefix?  
Would `WithRename` and `WithFilterAttributes` make sense?

Reply:
> Would `WithRename` and `WithFilterAttributes` make sense?

SGTM

Reply:
I did those two, do you want WithSetDescription, or is WithDescription good enough?

Reply:
> I did those two, do you want WithSetDescription, or is WithDescription good enough?

+1 on `WithSetDescription` to continue communicating the action that will be taken.